### DISCUSSION TITLE
Can I change the SWC parser’s behaviour to allow hyphens in identifiers?

### DISCUSSION THREAD

**Discussion by rinarakaki** [swc-project/swc#5523] [Q&A] [answered]
### Describe the feature

I believe the possible benefits of allowing hyphens everywhere in JS such as identifiers and fields of objects would easily surpass the cons of giving up calculating subtraction without spaces (`a-b`). So I’m wondering if we can have a setting or plugin to make this possible.

**Comment by kdy1** [comment]
Converted as this is not a feature.

**Comment by kwonoj** [comment, marked as answer]
SWC will not support non-ECMA syntaxes out of the box. It is up to your preference if you take some of the building blocks for the SWC from rust pacakges to invent new semantics.
